sklearn/model_selection/_split.py

```diff
@@ -287,9 +287,17 @@ def __init__(self, n_splits, shuffle, random_state):
             raise TypeError("shuffle must be True or False;"
                             " got {0}".format(shuffle))
 
+        if not shuffle and random_state is not None:  # None is the default
+            raise ValueError(
+                'Setting a random_state will have no effect when shuffle is '
+                'False.'
+            )
+
         self.n_splits = n_splits
         self.shuffle = shuffle
-        self.random_state = random_state
+        self.random_state = random_state  # only here for repr to work
+        self.seed = _get_seed_from_random_state_param(random_state)
```

Comment:
It's kinda odd to have a public attribute which is not set through `__init__` (I know it's not an estimator :p )

I think I'd be happier with `_seed`.

Reply:
No strong opinion here. I'll make it private